src/solvers/interior_point_Newton.jl

```diff
@@ -0,0 +1,385 @@
+    Hess_f_ = _to_mutating_gradient(Hess_f, evaluation)
+
+    g_ = isnothing(g) ? nothing : (M, p) -> g(M, p[])
+    grad_g_ = isnothing(grad_g) ? nothing : _to_mutating_gradient(grad_g, evaluation)
+    Hess_g_ = isnothing(Hess_g) ? nothing : _to_mutating_gradient(Hess_g, evaluation)
+    h_ = isnothing(h) ? nothing : (M, p) -> h(M, p[])
+    grad_h_ = isnothing(grad_h) ? nothing : _to_mutating_gradient(grad_h, evaluation)
+    Hess_h_ = isnothing(Hess_h) ? nothing : _to_mutating_gradient(Hess_h, evaluation)
+
+    rs = interior_point_Newton(
+        M,
+        f_,
+        grad_f_,
+        Hess_f_,
+        p_;
+        evaluation=evaluation,
+        g=g_,
+        h=h_,
+        grad_g=grad_g_,
+        grad_h=grad_h_,
+        Hess_g=Hess_g_,
+        Hess_h=Hess_h_,
+        kwargs...,
+    )
+    return (typeof(p_) == typeof(rs)) ? rs[] : rs
+end
+function interior_point_Newton(
+    M::AbstractManifold, cmo::O, p; kwargs...
+) where {O<:Union{ConstrainedManifoldObjective,AbstractDecoratedManifoldObjective}}
+    q = copy(M, p)
+    return interior_point_Newton!(M, cmo, q; kwargs...)
+end
+function interior_point_Newton!(
+    M::AbstractManifold,
+    f,
+    grad_f,
+    Hess_f,
+    p;
+    evaluation::AbstractEvaluationType=AllocatingEvaluation(),
+    g=nothing,
+    h=nothing,
+    grad_g=nothing,
+    grad_h=nothing,
+    Hess_g=nothing,
+    Hess_h=nothing,
+    inequality_constrains=nothing,
+    equality_constrains=nothing,
+    kwargs...,
+)
+    if isnothing(inequality_constrains)
+        inequality_constrains = _number_of_constraints(g, grad_g; M=M, p=p)
+    end
+    if isnothing(equality_constrains)
+        equality_constrains = _number_of_constraints(h, grad_h; M=M, p=p)
+    end
+    cmo = ConstrainedManifoldObjective(
+        f,
+        grad_f,
+        g,
+        grad_g,
+        h,
+        grad_h;
+        hess_f=Hess_f,
+        hess_g=Hess_g,
+        hess_h=Hess_h,
+        evaluation=evaluation,
+        equality_constrains=equality_constrains,
+        inequality_constrains=inequality_constrains,
+        M=M,
+        p=p,
+    )
+    dcmo = decorate_objective!(M, cmo; kwargs...)
+    return interior_point_Newton!(
+        M,
+        dcmo,
+        p;
+        evaluation=evaluation,
+        equality_constrains=equality_constrains,
+        inequality_constrains=inequality_constrains,
+        kwargs...,
+    )
+end
+function interior_point_Newton!(
+    M::AbstractManifold,
+    cmo::O,
+    p;
+    evaluation::AbstractEvaluationType=AllocatingEvaluation(),
+    X=get_gradient(M, cmo, p),
+    μ::Vector=ones(length(get_inequality_constraint(M, cmo, p, :))),
+    λ::Vector=zeros(length(get_equality_constraint(M, cmo, p, :))),
+    s=μ,
+    ρ=μ's / length(μ),
+    σ=calculate_σ(M, cmo, p, μ, λ, s),
+    γ=0.9,
+    retraction_method::AbstractRetractionMethod=default_retraction_method(M, typeof(p)),
+    sub_kwargs=(;),
+    _sub_M=M × Rn(length(λ)),
```

Comment:
```suggestion
    vector_space=Rn,
    _sub_M=M × vector_space(length(λ)),
```
This is more or less the key change for option 2 discussed above.

Reply:
...and Rn can stay as is? That would be neat and still fine with me :)

Reply:
Yes, Rn could stay as is. I think this would be the best solution based on our discussion :slightly_smiling_face: .

Reply:
And much easier than I already feared. Will do that tomorrow when I return to docs and tests for IPN (just finished the ConjugateRes part).

Reply:
Added that in all places where reasonable (in the high-level interface maybe not yet documented, tests for that are still not yet complete).